### PR TITLE
Add WebID-OIDC authentication method

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Implements issue #92.

Major changes:

* `solid.login()` and `solid.currentUser()` are switched over to use the WebID-OIDC auth client instead of WebID-TLS.
* TLS-based `login()` and `currentUser()` are still accessible via `solid.tls.login()` and `solid.tls.currentUser()`, though these will be deprecated in future versions.
* `solid.logout()` is implemented.

### CODE REVIEW

**Comment by dmitrizagidulin** [comment]
cc @dan-f / @timbl for review.

File: src/index.js [hunk 1]
```diff
@@ -50,13 +52,15 @@ const Solid = {
   AppRegistration: require('./solid/app-registration'),
   appRegistry: require('./app-registry'),
   auth,
+  tls,
   config: require('../config'),
   currentUser: auth.currentUser.bind(auth),
   identity: require('./identity'),
   login: auth.login.bind(auth),
```
Comment (dan-f) [inline src/index.js]: I didn't see any tests in `solid-auth-oidc` for the login function, so I don't like the idea of swapping out a tested `client.login` for an untested one.

File: src/index.js [hunk 2]
```diff
@@ -50,13 +52,15 @@ const Solid = {
...
+  logout: auth.logout.bind(auth),
```
Comment (dan-f) [inline src/index.js]: Same goes for logout - no tests.

File: webpack.config.js [hunk 3]
```diff
@@ -35,8 +35,12 @@ module.exports = {
...
   },
   externals: {
-    xhr2: 'XMLHttpRequest',
-    xmlhttprequest: 'XMLHttpRequest'
+    'xhr2': 'XMLHttpRequest',
+    'xmlhttprequest': 'XMLHttpRequest',
+    'node-fetch': 'fetch',
+    'text-encoding': 'TextEncoder',
+    'urlutils': 'URL',
+    'webcrypto': 'crypto'
```
Comment (dan-f) [inline webpack.config.js]: Are all these externals necessary? They're declared identically in [`solid-auth-oidc`'s webpack config](https://github.com/solid/solid-auth-oidc/blob/4cd999a1d4a7b0e763a2d90f179687f8df3a93e8/webpack.config.js).

File: .babelrc [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["es2015"]
```
Comment (dan-f) [inline .babelrc]: Mind also removing this in the `package.json` - https://github.com/solid/solid-client/blob/master/package.json#L27-L30

Comment (dmitrizagidulin) [inline .babelrc]: Sure thing, will remove.

File: src/index.js (hunk 1 above)
Comment (dmitrizagidulin) [inline src/index.js]: Right-o, I will be adding tests to both login & logout.

File: webpack.config.js (hunk 3 above)
Comment (dmitrizagidulin) [inline webpack.config.js]: I'd love to find a way to not include them; haven't been able to get it to work without.

File: src/index.js (hunk 1 above)
Comment (dmitrizagidulin) [inline src/index.js]: Added unit tests for login & logout, to `solid-auth-oidc`.

**Comment by dmitrizagidulin** [comment]
@dan-f Any other comments to address before this is merged?

**Comment by dan-f** [comment]
Let me take a look at those tests.

**Comment by dan-f** [comment]
Actually, is there a demo app to test this UI?

**Comment by dmitrizagidulin** [comment]
Sure, try http://solid.github.io/solid-oauth-oidc/demo/

**Comment by dan-f** [comment]
That page 404s

**Comment by dmitrizagidulin** [comment]
Argh, typo. https://solid.github.io/solid-auth-oidc/demo/

**Comment by dan-f** [comment]
Nice. So when I click "log out" and refresh the page it appears that I'm still logged in.

**Comment by dan-f** [comment]
I'm also seeing an XHR for a "/goodbye" HTML page which seems a bit strange

**Comment by dan-f** [comment]
So after further discussion, it looks like we'll namespace the new auth method separately rather than replace `solid.login` pending further integration testing

**Comment by dan-f** [comment]
ok :+1:

**Pull request merged**